x-pack/plugins/reporting/server/lib/store/store.ts

```diff
@@ -125,45 +145,152 @@ export class ReportingStore {
         browser_type: report.browser_type,
       },
     };
-    return this.client.callAsInternalUser('index', indexParams);
+    return await this.client.callAsInternalUser('index', indexParams);
   }
 
+  /*
+   * Called from addReport, which handles any errors
+   */
   private async refreshIndex(index: string) {
-    return this.client.callAsInternalUser('indices.refresh', { index });
+    return await this.client.callAsInternalUser('indices.refresh', { index });
   }
 
-  public async addReport(type: string, payload: unknown, options: AddReportOpts): Promise<Report> {
+  public async addReport(type: string, username: string | null, payload: unknown): Promise<Report> {
     const timestamp = indexTimestamp(this.indexInterval);
     const index = `${this.indexPrefix}-${timestamp}`;
     await this.createIndex(index);
 
     const report = new Report({
-      index,
+      _index: index,
       payload,
       jobtype: type,
-      created_by: options.created_by,
-      browser_type: options.browser_type,
-      max_attempts: options.max_attempts,
-      timeout: options.timeout,
-      priority: 10, // unused
+      created_by: username,
+      ...this.jobSettings,
     });
 
-    const doc = await this.saveReport(report);
-    report.updateWithDoc(doc);
+    try {
+      const doc = await this.indexReport(report);
+      report.updateWithEsDoc(doc);
 
-    await this.refreshIndex(index);
-    this.logger.info(`Successfully queued pending job: ${report._index}/${report.id}`);
+      await this.refreshIndex(index);
+      this.logger.info(`Successfully stored pending job: ${report._index}/${report._id}`);
 
-    return report;
+      return report;
+    } catch (err) {
+      this.logger.error(`Error in addReport!`);
+      this.logger.error(err);
+      throw err;
+    }
   }
 
-  public async updateReport(query: UpdateQuery): Promise<Report> {
-    return this.client.callAsInternalUser('update', {
-      index: query.index,
-      id: query.id,
-      if_seq_no: query.if_seq_no,
-      if_primary_term: query.if_primary_term,
-      body: { doc: query.body.doc },
-    });
+  /*
+   * Search for a report from task data and return back the report
+   */
+  public async findReport(taskJson: ReportTaskJSON): Promise<Report> {
```

Comment:
This is a new method to grab the "full" reporting doc when running the task, since Task Manager only holds the reference to the Report _id, _index and jobtype (for finding the approprate run function)